walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,69 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  private ArrayList<TimeRange> potentialTimes = new ArrayList<TimeRange>();
+  
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    potentialTimes.add(TimeRange.WHOLE_DAY);
+    if (request.getDuration() > 24 * 60){
+        return Arrays.asList();
+    }
+
+    //Collections.sort(events, TimeRange.ORDER_BY_START);
+    Set<String> requestAttendees = new HashSet<String>(request.getAttendees());
+
+    for(Event event : events){
+      HashSet<String> eventAttendees = new HashSet<String>();
+      eventAttendees.addAll(event.getAttendees());
```

Comment:
These two lines could be combined by passing `event.getAttendees()` as the argument to the HashSet constructor:` HashSet<String> eventAttendees = new HashSet<String>(event.getAttendees())`;